### DISCUSSION TITLE
Single Quick Bar

### DISCUSSION THREAD

**Discussion by matthiasdebaat** [home-assistant/frontend#11511] [Other feature requests]
The Quick Bar is an amazing functionality with lot of potential for every user. This prototype focuses on merging the two separate search bars for entities and commands. Currently you can open an entity filter (hotkey `E`) or the command pallet (hotkey `C`). In my opinion having two quick bars isn't necessary and we could merge them into a single one. Currently our quick bar searches on entities, pages and commands. In this design I've added some suggestions people can search for.

For now I've added a search icon on the dashboard to open it, but on another discussion I would like to investigate if it's nice to have a search icon on the sidebar.

[**Desktop prototype**](https://www.figma.com/proto/jqKXftriZ8gSQ0xVZLHYDf/Quick-bar?page-id=0%3A1&node-id=25%3A1592&viewport=241%2C48%2C0.47&scaling=min-zoom&starting-point-node-id=25%3A1592)
![CleanShot 2022-02-02 at 09 29 53](https://user-images.githubusercontent.com/20203284/152119164-dc506e9e-eb79-45db-8863-504001e7bdc2.gif)

[**Mobile prototype**](https://www.figma.com/proto/jqKXftriZ8gSQ0xVZLHYDf/Quick-bar?page-id=25%3A1307&node-id=25%3A1351&viewport=241%2C48%2C2.14&scaling=scale-down&starting-point-node-id=25%3A1351)
![CleanShot 2022-02-02 at 09 32 47](https://user-images.githubusercontent.com/20203284/152119462-5d0c0076-6597-4e2b-9c55-efe1649f408c.gif)

**Current Quick Bar**
<img width="240" alt="CleanShot 2022-02-02 at 09 34 23@2x" src="https://user-images.githubusercontent.com/20203284/152119693-f16ddeb8-eaee-4c4d-9ee0-2806dfddbbfa.png"> <img width="240" alt="CleanShot 2022-02-02 at 09 34 29@2x" src="https://user-images.githubusercontent.com/20203284/152119698-425d2f10-b8bb-44ba-8f79-7818152fb5d4.png">

**Comment by gabrielvisser** [comment]
Fantastic concept. I confess to barely using the quick bar(s) because having to remember to press a certain letter on the keyboard feels antithetical to an intuitive UX. Having it be a visible button AND combined seems like a logical evolution.

**Comment by dshokouhi** [comment]
I really dig how you have it being triggered now using a search button :) This will make mobile users happy and we can remove our workaround in the android app.  Currently today we trigger it using a 3 finger swipe down gesture but some devices use that gesture for other thing so not all users can use it. Now it will work across all setups :)

**Comment by emufan** [comment]
Like it. But please don't remove the hotkey(s).

BTW Wasn't it before one as well? At least I'm able to remove > in the one or add > in the other to search via versa.

**Reply by matthiasdebaat** [reply]
I agree that hotkeys should stay. Yeah, this redesign makes it possible to search all without the need of an `>`.

**Comment by BoneheadFraggle** [comment]
While I like the idea - I would also propose that all of the hotkeys, search functions like <https://github.com/home-assistant/frontend/pull/12324> and so on would allow us to turn them off completely for non-admin users.

I am my HAs only admin, and I like the features. But I do not want any of my HA users to find out about them. Then I will spend more time locking down HA then using it for new features.

I think "back-end information" in HA should be accessible on a need-to-know-basis. Which means I think it should be natural and easy to choose not to have access to Logbook and History (in more-info), hotkeys, global searches and so on. Therefore I would propose that all of this kind of functionality would be limited to admin users.

I would really like to have a panel for controlling the devices of my (the admins liking) without needing to reverse engineer solutions to keep other stuff from nosy kids, guests and other users. 
I think it would be a great selling point for HA to be able share HA with a guest/kids/partner without first needing to lock the system with a number of custom components and coding.
Now I realise what I am really after, is a "default kiosk mode", and that is probably a separate feature request.

Anyway, please allow all future shortcuts/quick find solutions to be easily turned off.

**Reply by emufan** [reply]
Yes and no. I don't like the ideas of the direction "the admin should do this and that". If there will be any improvement started anytime, then in an authorization concept and matrix. For each user, it should be selectable in admin console, which dashboards (including, history, log, etc. etc.) and which settings and shortcuts are allowed and are forbidden. But as said, please not hard-coded for a user type.

**Reply by matthiasdebaat** [reply]
As I can understand why you suggest this, I don't think it's the best default to disable search. Search is a really neat feature that all user should be able to use. Giving rights what kind of devices (or even entities) a user can control and what kind of pages it can visit is a complicated topic to implement.

**Reply by matthiasdebaat** [reply]
Maybe we can disable the `Command Palette` for non-admin users. Will check that.

**Reply by Mariusthvdb** [reply]
if anything, it would be nice if we could find it in the Dom path so we could card-mod-theme the magnifier button and set display: none on it.
Currently it seems to be non identifiable? There's no aria-label, nor title to select?

I tried but couldn't get to the button (or icon) solo, only both menu buttons, magnifier and 3-dots. see:

https://community.home-assistant.io/t/card-mod-super-charge-your-themes/212176/1079

**Comment by Mariusthvdb** [comment]
yes, please allow me to chime in, I hadnt seen this yet before I posted: https://github.com/home-assistant/frontend/discussions/12384

there is a tendency to see more and more Tips and Magnifiers, which is not really useful for the overall experience by default per se. I for one don't care for the tip we now see in  dev tools Set state,(I am there so dont need the tip) nor for the Magnifier in all of my carefully curated frontend Menu bars... Nice to have on keystroke, but not 'always on'.

Especially not for the non-admin users, who in my household should never interfere with any backend search. So yeah, I like the search functionality. A lot. But not always on, and Not for everyone.
https://github.com/home-assistant/frontend/discussions/12022#discussioncomment-2627542

**Reply by matthiasdebaat** [reply]
As replied here https://github.com/home-assistant/frontend/discussions/11511#discussioncomment-2628593

We're looking into having a single search in the sidebar, instead of magnify icons on different pages.

**Reply by Mariusthvdb** [reply]
that would already cleanup the views very nicely, thanks for considering that
